File: project/angular_project/src/app/models/sensor.ts
import { Type } from "./type";

export class Sensor {
  sensor_id: number;
  value: number;
  version: string;
  type: Type;

  constructor() {
    this.sensor_id = 0;
    this.value = 0;
    this.version = '';
    this.type = new Type(); 
  }
}

File: project/src/main/java/com/example/project/entities/Sensor.java
package com.example.project.entities;

import java.io.Serializable;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;


@Data
@Document(collection = "Sensor")
public class Sensor implements Serializable {
	@Id
	private int sensor_id;
	private int value;
	private String version;
	private Type type;
	
}